dask/array/core.py

```diff
@@ -2995,7 +3009,10 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
                "\nData has %d dimensions, but got axis=%d")
         raise ValueError(msg % (ndim, axis))
 
-    if n == 1:
+    n = len(seq)
+    if n == 0:
+        return from_array(np.empty(shape=shape, dtype=meta.dtype))
```

Comment:
This isn't very meta-friendly. Fixing in PR ( https://github.com/dask/dask/pull/4979 ).